### PR TITLE
hibernate.ddl-validate; Liquibase; BookRepository; BookController; BookService:

### PR DESCRIPTION
**Pull request by DEz-1996**
Replaced BookRepository with use JpaRepository interface
Added Liquibase support
Added liquibase change file: 01-create-books-table.yaml
Changed spring.jpa.hibernate.ddl-auto property value to validate
Implemented soft delete concept for the Book entity
Added "update" and "delete" endpoints in BookController
Added "update" and "deleteById" methods for BookService

### CODE REVIEW

File: src/main/java/online/book/store/controller/BookController.java [hunk 1]
```diff
@@ -5,21 +5,25 @@
...
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-@RequestMapping("/books")
+@RequestMapping("/api/books")
```
Comment (ponomvrenko) [inline src/main/java/online/book/store/controller/BookController.java]: `server.servlet.context-path=/api` use it in your applications.properties file

File: src/main/java/online/book/store/controller/BookController.java [hunk 2]
```diff
@@ -5,21 +5,25 @@
...
 @RequiredArgsConstructor
 public class BookController {
 
     private final BookService bookService;
 
-    @GetMapping
+    @GetMapping()
```
Comment (ponomvrenko) [inline src/main/java/online/book/store/controller/BookController.java]: Round brackets are not needed here

File: src/main/java/online/book/store/controller/BookController.java [hunk 3]
```diff
@@ -29,8 +33,19 @@ public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBook(id);
     }
 
-    @PostMapping
+    @PostMapping()
```
Comment (ponomvrenko) [inline src/main/java/online/book/store/controller/BookController.java]: Same situation

File: src/main/java/online/book/store/repository/BookRepository.java [hunk 4]
```diff
@@ -1,13 +1,7 @@
...
 import online.book.store.model.Book;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    Optional<Book> findBookById(Long id);
-
-    List<Book> findAll();
+public interface BookRepository extends JpaRepository<Book, Long> {
```
Comment (ponomvrenko) [inline src/main/java/online/book/store/repository/BookRepository.java:8]: Add Repository annotation here

**Review by ponomvrenko — CHANGES_REQUESTED** [review]

**Comment by ponomvrenko** [comment]
Hello 😁 Well done! I just left some minor points that will improve a little bit your code

File: src/main/java/online/book/store/repository/BookRepository.java (hunk 4 above)
Comment (vovasalyha) [reply src/main/java/online/book/store/repository/BookRepository.java:8]: It's not necessary for interfaces which extend Springs JPA related interfaces like `JpaRepository` and Spring applies it by default.

File: src/main/java/online/book/store/service/imp/BookServiceImpl.java [hunk 5]
```diff
@@ -31,9 +31,17 @@ public BookDto save(CreateBookRequestDto bookDto) {
         return bookMapper.toDto(savedBook);
     }
 
+    @Override
+    public BookDto update(CreateBookRequestDto bookDto, Long id) {
```
Comment (vovasalyha) [inline src/main/java/online/book/store/service/imp/BookServiceImpl.java:35]: It's good to check if this ID exists to throw an exception with proper message if not.

**Review by vovasalyha — APPROVED** [review]
Approving, but please take a look on my comment regarding update method ID validation.

File: src/main/java/online/book/store/repository/BookRepository.java (hunk 4 above)
Comment (ponomvrenko) [reply src/main/java/online/book/store/repository/BookRepository.java:8]: Oh, I see.. my bad

**Pull request merged**